### PR TITLE
[Kotlin Migration] RustDroidBackend 

### PR DESCRIPTION
**Pull request by codingtosh**
[Kotlin Migration] RustDroidBackend

### CODE REVIEW

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by david-allison-1 — APPROVED** [review]
LGTM

File: AnkiDroid/src/main/java/com/ichi2/libanki/backend/RustDroidBackend.kt [hunk 1]
```diff
@@ -13,113 +13,87 @@
...
+import com.ichi2.libanki.TemplateManager.TemplateRenderContext
+import com.ichi2.libanki.backend.exception.BackendNotSupportedException
+import com.ichi2.libanki.backend.model.SchedTimingToday
+import com.ichi2.libanki.backend.model.SchedTimingTodayProto
+import com.ichi2.libanki.utils.Time
+import net.ankiweb.rsdroid.BackendFactory
+import net.ankiweb.rsdroid.database.RustV11SQLiteOpenHelperFactory
+
+/** The V11 Backend in Rust  */
+open class RustDroidBackend( // I think we can change this to BackendV1 once new DB() accepts it.
```
[the remark this answers is not in the document]
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/libanki/backend/RustDroidBackend.kt:32]: The comment is in the primary constructor of the class so it points to the properties in it, not the class itself. 

I had to change `mBackend `to `backend `due to kotlin lint checks, but  the property backend(which was mBackend earlier) is at the line just below the comment. (34)
 `private val backend : BackendFactory`

Should I move the comment down for better readability?

File: AnkiDroid/src/main/java/com/ichi2/libanki/backend/RustDroidBackend.kt [hunk 2]
```diff
@@ -13,113 +13,87 @@
...
+    private val backend: BackendFactory
+) : DroidBackend {
+    override fun createCollection(context: Context, db: DB, path: String?, server: Boolean, log: Boolean, time: Time): Collection {
```
[the remark this answers is not in the document]
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/libanki/backend/RustDroidBackend.kt:35]: Sure, understood.

[the remark this answers is not in the document]
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/backend/RustDroidBackend.kt:35]: #10171 just merged

[the remark this answers is not in the document]
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/libanki/backend/RustDroidBackend.kt:35]: @mikehardy Thanks for the heads up, will rebase and amend the commits later today.

**Review by mikehardy — APPROVED** [review]
This looks fine to me, I'd merge it - then I think it unblocks a different PR... @Arthur-Milchior  ?

**Comment by codingtosh** [comment]
> This looks fine to me, I'd merge it - then I think it unblocks a different PR... @Arthur-Milchior ?

@mikehardy I think this can merge as I had moved the comment down to the line below as per @Arthur-Milchior 's [comment](https://github.com/ankidroid/Anki-Android/pull/10143#discussion_r783205887).

**Comment by mikehardy** [comment]
Yeah, I'm just doing local compile testing on current PRs now and was going to merge this one. Thanks @codingtosh

**Pull request merged**